reference_validator/YAML_HotValidator.py

```diff
@@ -0,0 +1,724 @@
+                    env_node.structure = yaml.load(fd.read())
+            except IOError:
+                print('File ' + env_node.path + ' could not be opened.')
+                sys.exit(1)
+
+            # Add to currently validated files
+            self.curr_nodes.insert(0, env_node)
+
+            # Save mappings
+            if 'resource_registry' in env_node.structure:
+                for origin, custom in six.iteritems(env_node.structure['resource_registry']):
+                    if type(custom) == str:
+                        env_node.resource_registry[origin] = custom
+                    elif origin == 'resources':
+
+                        # Find if there is any mapping (hooks etc are not important)
+                        for res in env_node.structure['resource_registry']['resources'].keys():
+                            for key, value in six.iteritems(
+                                env_node.structure['resource_registry']['resources'][res]):
+
+                                # Add indirect mapping using regexp - multiple indentations
+                                if (type(value) == str) and (value.endswith('.yaml')):
+                                    env_code.resource_registry[key] = [value, res]
+
+            # Save additional parameters + parameters with default values
+            if 'parameters' in env_node.structure:
+                for par in list(env_node.structure['parameters'].keys()):
+                    env_node.params[par] = False
+
+            if 'parameter_defaults' in env_node.structure:
+                for par in list(env_node.structure['parameter_defaults'].keys()):
+                    env_node.params_default[par] = False
+
+            # Create HOT files with mapped files
+            for child in list(env_node.resource_registry.values()):
+                if ((type(child) == str and child.endswith('.yaml')) or
+                    ((type(child) == list) and child[0].endswith('.yaml'))):
+
+                    # If a file is created already as a root, no need for redundancy
+                    found = False
+                    for m in self.mappings:
+                        if ((m.path == child) and (m.parent in self.environments)):
+                            env_node.children.insert(0, m)
+                            found = True
+                            break
+
+                    if not found:
+                        env_node.children.insert(0, HOT.YAML_Hotfile(env_node, child))
+                        self.mappings.append(env_node.children[0])
+
+            # Remove from currently validated files
+            self.curr_nodes.remove(env_node)
+
+
+    def add_param_defaults(self):
+        ''' Add default from param_defaults where missing '''
+        for hot in self.templates + self.mappings:
+            for p in hot.params:
+
+                # Try to find param_default
+                if p.default is None:
+                    flag = False
```

Comment:
Could you rename `flag` to something more descriptive? It's not really clear what it's for now -- it may just have easily been called `boolean` or `data`.